lib/consumer/amqp_data_consumer_supervisor.ex

```diff
@@ -3,30 +3,42 @@
 
 defmodule Mississippi.Consumer.AMQPDataConsumer.Supervisor do
   @moduledoc false
-  use Supervisor
+  use Horde.DynamicSupervisor
 
+  alias Horde.DynamicSupervisor
   alias Mississippi.Consumer.AMQPDataConsumer
 
+  require Logger
+
   def start_link(init_arg) do
-    Supervisor.start_link(__MODULE__, init_arg, name: __MODULE__)
+    DynamicSupervisor.start_link(__MODULE__, init_arg,
+      name: __MODULE__,
+      distribution_strategy: Horde.UniformQuorumDistribution
+    )
   end
 
   @impl true
-  def init(init_arg) do
-    children =
-      amqp_data_consumers_childspecs(init_arg[:queues_config])
+  def init(_init_arg) do
+    DynamicSupervisor.init(
+      members: :auto,
+      strategy: :one_for_one,
+      process_redistribution: :active
+    )
+  end
 
-    opts = [strategy: :one_for_one]
+  def start_children(queues_config) do
+    children = amqp_data_consumers_childspecs(queues_config)
 
-    Supervisor.init(children, opts)
+    Enum.each(children, fn child ->
+      DynamicSupervisor.start_child(Mississippi.Consumer.AMQPDataConsumer.Supervisor, child)
+    end)
   end
 
   defp amqp_data_consumers_childspecs(queues_config) do
-    queue_range_start = queues_config[:range_start]
-    queue_range_end = queues_config[:range_end]
+    queue_total = queues_config[:total_count]
     queue_prefix = queues_config[:prefix]
 
-    for queue_index <- queue_range_start..queue_range_end do
+    for queue_index <- 0..(queue_total - 1) do
```

Comment:
```suggestion
    max_index = queue_total - 1

    for queue_index <- 0..max_index do
```

nit: makes the code easier on the eyes IMO